zig-mode.el

```diff
@@ -117,6 +128,66 @@ If given a SOURCE, execute the CMD on it."
   (interactive)
   (zig--run-cmd "run" (file-local-name (buffer-file-name)) "-O" zig-run-optimization-mode))
 
+;; zig flymake
+
+(defvar zig--flymake-proc nil
+  "Current zig flymake process.")
+(defun zig-flymake (report-fn &rest _args)
+  "Zig flymake command, called by flymake-start."
+  (unless (executable-find "zig")
+    (error "Cannot find suitable zig executable"))
+  ;; Kill the last zig--flymake-proc, as flymake can spawn them indefinitely.
+  (when (process-live-p zig--flymake-proc)
+    (kill-process zig--flymake-proc))
+
+  (save-restriction
+    (widen)
+    (setq
+     zig--flymake-proc
+     (make-process
+      :name "zig-flymake" :noquery t :connection-type 'pipe
+      :buffer (generate-new-buffer "*zig-flymake*")
```

Comment:
Worth checking here whether this honours buffer-local `exec-path` (ie. `$PATH`) settings in the initial source buffer, which is necessary when using `direnv` and `envrc.el` to put a specific `zig` on the path.